cmd/snap/cmd_connections.go

```diff
@@ -0,0 +1,222 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+
+	"github.com/jessevdk/go-flags"
+)
+
+type cmdConnections struct {
+	clientMixin
+	All          bool `long:"all"`
+	Disconnected bool `long:"disconnected"`
+	Positionals  struct {
+		Snap installedSnapName `skip-help:"true"`
+	} `positional-args:"true"`
+}
+
+var shortConnectionsHelp = i18n.G("List interface connections")
+var longConnectionsHelp = i18n.G(`
+List connections between plugs and slots. When passed an optional
+snap name, lists connected and unconnected plugs and slots for
+that snap only.
+
+Pass --all to list connected and unconnected plugs and slots.
+`)
+
+func init() {
+	addCommand("connections", shortConnectionsHelp, longConnectionsHelp, func() flags.Commander {
+		return &cmdConnections{}
+	}, map[string]string{
+		"all":          i18n.G("Show connected and unconnected plugs and slots"),
+		"disconnected": i18n.G("Show disconnected plugs and slots"),
+	}, []argDesc{{
+		// TRANSLATORS: This needs to be wrapped in <>s.
+		name: i18n.G("<snap>"),
+		// TRANSLATORS: This should not start with a lowercase letter.
+		desc: i18n.G("Constrain listing to a specific snap"),
```

Comment:
I think descriptions should be full sentences, but I know they currently aren't, everywhere. We're inconsistent. We should discuss it and fix it sometime.